include/os/linux/kernel/linux/simd_powerpc.h

```diff
@@ -83,6 +84,20 @@
 		disable_kernel_altivec();	\
 		preempt_enable();		\
 	}
+#else /* CONFIG_SPE */
+#define	kfpu_begin()				\
+	{					\
+		preempt_disable();		\
+		enable_kernel_altivec();	\
+		enable_kernel_vsx();		\
+	}
+#define	kfpu_end()				\
+	{					\
+		disable_kernel_vsx();		\
+		disable_kernel_altivec();	\
+		preempt_enable();		\
+	}
+#endif
 #else
 /* seems that before 4.5 no-one bothered */
 #define	kfpu_begin()
```

Comment:
Does anyone know why on pre-4.5 kernels kfpu_begin() is a no-op while kfpu_end() calls preempt_enable()? This feels wrong to me. :/

Reply:
It does feel wrong, however my recollection is that this was simply the way it was done in these older kernels.  There was some discussion of this in https://github.com/openzfs/zfs/pull/9539.